pkg/services/annotations/annotationsimpl/loki/historian_store.go

```diff
@@ -78,6 +78,11 @@ func (r *LokiHistorianStore) Get(ctx context.Context, query *annotations.ItemQue
 		return make([]*annotations.ItemDTO, 0), nil
 	}
 
+	// alert state history does not contain tags, so we should return nothing if we want to match all tags from the query
+	if len(query.Tags) > 0 && !query.MatchAny {
```

Comment:
What happens if the query contains tags, but MatchAny is true? Does MatchAny still imply it needs to match at least one tag in the list?

If so, I'd imagine we want to return nothing in that case too, since it can never match any specific tag either?

Reply:
The UI description for the flag is `Enabling this returns annotations that match any of the tags specified below`.

So I think you're right, it's basically an "any vs all" toggle and state history won't match either case since it's never going to have tags.